### PR TITLE
Feature/delete sheets 98674254

### PR DESCRIPTION
**Pull request by sidke**
delete button:
image

invalid deletion:
image

valid deletion:
image

booted off sheet by deletion:
image

### CODE REVIEW

**Comment by nfiniteset** [comment]
We have a typo in the 'valid deletion' example.

**Comment by nfiniteset** [comment]
Also the second example lacks a period after 'Once deleted you will not be able to get it back' but all the other statements have punctuation.

**Comment by sidke** [comment]
the valid deletion isn't a type, it's the sheet name. i was pulling off of the design with the sheet name being on a new line 
image

**Comment by sidke** [comment]
i am also waiting for different copy for being booted off a sheet, even though it's perfect as is

**Comment by nfiniteset** [comment]
Ah that makes sense @sidke. Thanks for entertaining my out-of-nowhere comments on this. Still missing that period though :wink: 

Re: getting booted, I can't imagine how that could be improved. :+1:

**Comment by sidke** [comment]
made the screenshot before adding the period, this is from the last commit
image

File: assets/css/modules/header.less [hunk 1]
```diff
@@ -237,7 +238,7 @@ nav#header {
       .transform(translate(-240px, 0));
     }
 
-    .icon-menu, .icon-download, .icon-disguise, .icon-logout, .icon-stickies, .icon-cleanup {
+    [class*="icon-"] {
```
Comment (njackson) [inline assets/css/modules/header.less]: :clap: :clap:

File: assets/js/views/feature_menu.coffee [hunk 2]
```diff
@@ -53,3 +55,24 @@ class boardroom.views.FeatureMenu extends boardroom.views.Base
...
+    else
+      new boardroom.views.Modal
+        class: "modal-delete"
+        title: "Are you sure you want to delete <br /> #{@model.activeSheet().name()}?"
+        message: "Once deleted you will not be able to get it back."
+        buttons:
+          "Delete": (modal) =>
+            @model.activeSheet().delete()
+            modal.close()
+          "Cancel": (modal) =>
```
Comment (njackson) [inline assets/js/views/feature_menu.coffee]: how can these be made to differentiate between primary/secondary or destructive actions? feels like we'll need to be able to also give classes to the buttons. this is good for now, just food for thought

File: assets/js/views/modal.coffee [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    @buttons = []
+    for caption, button of attributes.buttons
+      if (button instanceof Function)
+        button =
+          callback: button
+      _.extend button,
+        caption: caption
+        class: caption.toLowerCase()
+      @buttons.push(button)
+    @buttons[@buttons.length - 1].class += " primary"
```
Comment (sueyeh) [inline assets/js/views/modal.coffee]: does this set the color of the last button?

File: assets/js/views/feature_menu.coffee (hunk 2 above)
Comment (sidke) [inline assets/js/views/feature_menu.coffee]: you can style them specifically by their modal class and button class, as in: `#modal-box.modal-delete .button.cancel` or `.button.delete` but for now i just opted to make the last button in the modal the red one

File: assets/js/views/modal.coffee (hunk 3 above)
Comment (sidke) [inline assets/js/views/modal.coffee]: yes

**Comment by sueyeh** [comment]
lgtm

File: assets/js/views/modal.coffee [hunk 4]
```diff
@@ -0,0 +1,59 @@
+class boardroom.views.Modal extends Backbone.View
+  el: '#modal'
+
+  template: _.template '''
+    <div id="modal-container">
+      <div id="modal-backdrop">
```
Comment (njackson) [inline assets/js/views/modal.coffee]: Most modal implementations put the backdrop on the very bottom of the page AFAIK. I vaguely recall something about strange click/scroll interactions and/or z-index issues.

**Comment by njackson** [comment]
LGTM

**Pull request merged**